mypy/checkstrformat.py

```diff
@@ -151,13 +151,11 @@ def check_simple_str_interpolation(self, specifiers: List[ConversionSpecifier],
 
     def check_mapping_str_interpolation(self, specifiers: List[ConversionSpecifier],
                                        replacements: Expression) -> None:
-        if (isinstance(replacements, DictExpr) and
-                all(isinstance(k, (StrExpr, BytesExpr))
-                    for k, v in replacements.items)):
+        if isinstance(replacements, DictExpr):
```

Comment:
This is generally not right. Consider this example:
```py
x = 'q'
'%(q)d' % {'a': 1, x: 2}   # actually okay
```
What about still using the `all(...)` check (include also `unicode`), but only for the `key_not_in_mapping` error? Item types can still be checked even if some keys are not literals -- we can assume that there are no duplicate keys.